src/curry/debounce.ts

```diff
@@ -37,7 +37,7 @@ export type DebounceFunction<TArgs extends any[]> = {
  * ```
  */
 export function debounce<TArgs extends any[]>(
-  { delay }: { delay: number },
+  { delay, leading = false }: { delay: number; leading?: boolean },
```

Comment:
in this case it would be of great value to add a JsDoc

```ts 
export function debounce<TArgs extends any[]>(
  { delay, leading = false }: { 
    delay: number;
    /**
     * Decides whether the source function is called on the first
     * invocation of the throttle function or not
     * 
     * @default false
     */
    leading?: boolean
  },
  func: (...args: TArgs) => any,
): DebounceFunction<TArgs> {
```
![image](https://github.com/user-attachments/assets/093bec9a-320b-4ae7-8bfc-7fb6f8e362fe)

Reply:
Definitely. Added.